test/integration/logging_test.go

```diff
@@ -0,0 +1,236 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}, metav1.CreateOptions{})
+	if errNamespace != nil {
+		t.Fatal(errNamespace)
+	}
+	// This register the function.
+	t.Cleanup(func() {
+		// delete the test namespace
+		errNamespace = clientset.CoreV1().Namespaces().Delete(context.Background(), namespaceName, metav1.DeleteOptions{})
+		if errNamespace != nil {
+			t.Fatal(errNamespace)
+		}
+	})
+
+	_, roleBindingErr := clientset.RbacV1().RoleBindings(namespaceName).Create(context.Background(), &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "view-default-test",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      "default",
+				Namespace: namespaceName,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "view",
+		},
+	},
+		metav1.CreateOptions{})
+	if roleBindingErr != nil {
+		t.Fatal(roleBindingErr)
+	}
+
+	token, tokenErr := clientset.CoreV1().ServiceAccounts(namespaceName).CreateToken(context.Background(), "default", &authenticationv1.TokenRequest{}, metav1.CreateOptions{})
+	if tokenErr != nil {
+		fmt.Printf("could not create a token, %s", tokenErr)
+		t.Fatal(tokenErr)
+	}
+
+	clientHTTP := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	kac := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "KubeArchiveConfig",
+			"apiVersion": fmt.Sprintf("%s/%s", kubearchivev1alpha1.SchemeBuilder.GroupVersion.Group, kubearchivev1alpha1.SchemeBuilder.GroupVersion.Version),
+			"metadata": map[string]string{
+				"name":      "kubearchive",
+				"namespace": namespaceName,
+			},
+			"spec": map[string]any{
+				"resources": []map[string]any{
+					{
+						"selector": map[string]string{
+							"apiVersion": "v1",
+							"kind":       "Pod",
+						},
+						"archiveWhen": "true",
+						"deleteWhen":  "status.phase == 'Succeeded'",
+					},
+				},
+			},
+		},
+	}
+
+	gvr := kubearchivev1alpha1.GroupVersion.WithResource("kubearchiveconfigs")
+	_, kacErr := dynamicClient.Resource(gvr).Namespace(namespaceName).Create(context.Background(), kac, metav1.CreateOptions{})
+	if kacErr != nil {
+		t.Fatal(kacErr)
+	}
+
+	// Give a bit of time to KubeArchive to pickup filters changes
+	time.Sleep(5 * time.Second)
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fedora",
+			Namespace: namespaceName,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Containers: []corev1.Container{
+				corev1.Container{
+					Name:    "fedora",
+					Command: []string{"echo", "Hello World"},
+					Image:   "quay.io/fedora/fedora:latest",
+				},
+			},
+		},
+	}
+
+	_, createPodErr := clientset.CoreV1().Pods(namespaceName).Create(context.Background(), pod, metav1.CreateOptions{})
```

Comment:
This is the simplest case of the logging retrieval, retrieving the logs of a pod, and although I think that this is a nice test to have, I miss being able to also test the more "complex" case of retrieving the logs from an owner of the pod (like we were doing with the `CronJob`  when dealing with the cronjob-generator)

Another alternative could be to create a deployment with one pod, that's simpler than the cronjob and if we ask for the logs of the deployment it would be the same as askinf for the pods of the cronjob, a two-level ownership jump.

Reply:
I agree, I'm experimenting now to make Splunk go faster, then I will switch the approach to some resource with children. Maybe a Job will be better, so we don't need to stop it before retrieving logs.

Reply:
I changed to use the cron generator